frontend/src/queries/nodes/DataVisualization/displayLogic.ts

```diff
@@ -0,0 +1,95 @@
+import * as d3 from 'd3'
+import { actions, afterMount, connect, kea, key, path, props, reducers } from 'kea'
+import { subscriptions } from 'kea-subscriptions'
+
+import { GoalLine } from '~/queries/schema'
+
+import { dataVisualizationLogic } from './dataVisualizationLogic'
+import type { displayLogicType } from './displayLogicType'
+
+export interface DisplayLogicProps {
+    key: string
+}
+
+export const displayLogic = kea<displayLogicType>([
+    key((props) => props.key),
+    path(['queries', 'nodes', 'DataVisualization', 'displayLogic']),
+    props({ key: '' } as DisplayLogicProps),
+    connect({
+        values: [dataVisualizationLogic, ['yData', 'query']],
+        actions: [dataVisualizationLogic, ['setQuery']],
+    }),
+    actions(({ values }) => ({
+        addGoalLine: () => ({ yData: values.yData }),
+        updateGoalLine: (goalLineIndex: number, key: string, value: string | number) => ({
+            goalLineIndex,
+            key,
+            value,
+        }),
+        removeGoalLine: (goalLineIndex: number) => ({ goalLineIndex }),
+        setGoalLines: (goalLines: GoalLine[]) => ({ goalLines }),
+    })),
+    reducers({
+        goalLines: [
+            [] as GoalLine[],
+            {
+                addGoalLine: (state, { yData }) => {
+                    const yDataFlat = yData?.flatMap((n) => n.data) ?? []
+                    const yDataAvg = Math.round(d3.mean(yDataFlat) ?? 0)
+
+                    return [
+                        ...state,
+                        {
+                            label: 'Q4 Goal',
+                            value: yDataAvg ?? 0,
+                        },
+                    ]
+                },
+                removeGoalLine: (state, { goalLineIndex }) => {
+                    const goalLines = [...state]
+
+                    goalLines.splice(goalLineIndex, 1)
+                    return goalLines
+                },
```

Comment:
An easier way to do this is `return state.filter((elem, index) => index !== goalLineIndex)`